### ISSUE TITLE
Generate cjklib database on first use

### ISSUE THREAD

**Issue by chatch** [chatch/pinyin-toolkit#5]
In Ank1 the cjklib was generated on Anki startup if it was the first run of the plugin or a dictionary had been updated. There was a tricky workaround to allow this without locking problems. Comment from main.py as follows:

```
    # Right, this is more than a bit weird. The basic issue is that if we were
    # to just initialize() RIGHT NOW then we will hold the Python import lock,
    # because Anki __import__s its plugins. This ABSOLUTELY KILLS us when we
    # come to e.g. build the database on a background thread, because that code
    # naturally wants to import some stuff, but it doesn't hold the lock!
    #
    # To work around this issue, we carefully schedule initialization (and database
    # construction) for a time when Anki will not have caused us to hold the import lock.
```

Now in Anki2 I can't seem to workaround this ... The app is locking. The popup dialog "rebuilding the database etc." is displayed but the database is not being built. When the window is called Anki starts and only then does the database build thread kick off and run.

So ... At the moment I do a manual build of the database before Anki is run.

This should change to do rebuilds as before. However i think it can happen on first use of the plugin functionality rather then on Anki startup. Eg. User hits Anki prefs, or edits a card, basically any of the hooks get triggered.

**Comment by Crissman** [comment]
I think I'm seeing this issue as well on a Mac.  Every time I load I see the "Pinyin Toolkit is rebuilding its database of Chinese readings and meaings."

How do you manually build the DB before Anki is run?

**Comment by chatch** [comment]
you can run "python pinyin/db/builder.py" from the command line. 

I did this manually and bundled the built database with the plugin. So that's a reason it's a much larger download then in PTK 1.

**Comment by Crissman** [comment]
No luck.

Crissmans-iMac:addons Crissman$ python pinyin/db/builder.py
Traceback (most recent call last):
  File "pinyin/db/builder.py", line 9, in <module>
    from pinyin.logger import log
ImportError: No module named pinyin.logger

**Comment by chatch** [comment]
can you try touching the database and restarting Anki. See if it rebuilds the database or not. From command line:

> touch ./pinyin/db/cjklib.db

Let me know if that works. Because a timestamp compare is done on startup. If the database has the most recent timestamp it shouldn't rebuild.

btw. In a month or so i'm going to have access to a friends mac so I'll give the plugin a run and fix some issues then. unless there's a mac / python dude that can take a look in the meantime. anyone..!?

**Comment by Crissman** [comment]
Yup, that seems to have fixed the "Rebuilding the DB" message everytime I
started Anki.  Problem solved?

**Comment by chatch** [comment]
yeah i guess for this case. still the original problem of this Issue stands.
